dubbo-rpc/dubbo-rpc-dubbo/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/InvokerTelnetHandlerTest.java

```diff
@@ -69,6 +66,21 @@ public void testInvokeDefaultSService() throws RemotingException {
         assertTrue(result.contains("result: \"ok\""));
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testInvokeWithSpecifyService() throws RemotingException {
+        mockChannel = mock(Channel.class);
+        given(mockChannel.getAttribute("telnet.service")).willReturn(null);
+        given(mockChannel.getLocalAddress()).willReturn(NetUtils.toAddress("127.0.0.1:5555"));
+        given(mockChannel.getRemoteAddress()).willReturn(NetUtils.toAddress("127.0.0.1:20886"));
+
+        ProviderModel providerModel = new ProviderModel("org.apache.dubbo.rpc.protocol.dubbo.support.DemoService", new DemoServiceImpl(), DemoService.class);
```

Comment:
I think 
    `ProviderModel providerModel = new ProviderModel(DemoService.class.getName(), new DemoServiceImpl(), DemoService.class);` would better approach, it will avoid the the problem if class package changes in future. What would you say?